compliance/collection/main.go

```diff
@@ -154,18 +159,24 @@ func manageSendToSensor(ctx context.Context, cli sensor.ComplianceService_Commun
 	}
 }
 
-func manageNodeScanLoop(ctx context.Context, rescanInterval time.Duration, scanner nodeinventorizer.NodeInventorizer) <-chan *sensor.MsgFromCompliance {
-	sensorC := make(chan *sensor.MsgFromCompliance)
+func manageNodeScanLoop(ctx context.Context, rescanInterval time.Duration, scanner nodeInventorizer) <-chan *sensor.MsgFromCompliance {
 	nodeName := getNode()
+	sensorC := make(chan *sensor.MsgFromCompliance)
+
 	go func() {
 		defer close(sensorC)
 		t := time.NewTicker(rescanInterval)
 
 		// first scan should happen on start
-		msg, err := scanNode(nodeName, scanner)
+		inventory, err := scanner.Scan(nodeName)
 		if err != nil {
-			log.Errorf("error running scanNode: %v", err)
+			log.Errorf("error running cachedScanNode: %v", err)
 		} else {
+			msg := &sensor.MsgFromCompliance{
+				Node: nodeName,
+				Msg:  &sensor.MsgFromCompliance_NodeInventory{NodeInventory: inventory},
+			}
+			cmetrics.ObserveInventoryProtobufMessage(msg)
 			sensorC <- msg
 		}
```

Comment:
This fragment keeps growing. I think it is maybe a time to consider a nested function (warning: untested)

```go
func manageNodeScanLoop(ctx context.Context, rescanInterval time.Duration, scanner nodeInventorizer) <-chan *sensor.MsgFromCompliance {
	nodeName := getNode()
	sensorC := make(chan *sensor.MsgFromCompliance)
	
	scanFunc := func() {
		inventory, err := scanner.Scan(nodeName)
		if err != nil {
			log.Errorf("error running cachedScanNode: %v", err)
		} else {
			msg := &sensor.MsgFromCompliance{
				Node: nodeName,
				Msg:  &sensor.MsgFromCompliance_NodeInventory{NodeInventory: inventory},
			}
			cmetrics.ObserveInventoryProtobufMessage(msg)
			sensorC <- msg
		}
	}

	go func() {
		defer close(sensorC)
		t := time.NewTicker(rescanInterval)

		// first scan should happen on start
		scanFunc()

		for {
			select {
			case <-ctx.Done():
				return
			case <-t.C:
				scanFunc()
			}
		}
	}()
	return sensorC
}
```

Reply:
Done!